feat: Updates User List to support created_at & last_active_at date filters

BrandonRomano opened the pull request:
Add `CreatedAtBefore`, `CreatedAtAfter`, `LastActiveAtBefore` and `LastActiveAtAfter` to `user.ListParams`.

BrandonRomano committed:
Updates User List to support created_at & last_active_at date filters

BrandonRomano committed:
Adds Changelog entry

gkats commented on CHANGELOG.md:
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 2.1.1
```
:100: Thank you for adding a changelog entry!

:question: Should we avoid adding a header for the version until we actually release it?

The next version bump could also be minor (not patch). Is it better if we leave the entries at the top without a header? I guess it doesn't matter much, but it would be good to follow a convention among us.

Feel free to leave as is.

BrandonRomano replied:
> ❓ Should we avoid adding a header for the version until we actually release it?

You know, this is a good idea.  I thought of this exact problem when I was typing up this header.

Perhaps we can use the header `## Next release`, just so it's contextualized to anyone who peeks the changelog?

gkats reviewed (approved):
:rocket:

BrandonRomano committed:
Update header to be next release

This work has yet to be released. When we cut a release, we will update
the changelog header to be the release number.
